### PR TITLE
Update Features.md

### PR DESCRIPTION
**Pull request by surendrapathak**
Improve documentation of rules

### CODE REVIEW

File: Features.md [hunk 1]
```diff
@@ -14,27 +14,308 @@
...
+If you have ideas for additions or new detection techniques,
+please [contribute](https://github.com/interlynk-io/sbomqs#contributions)!
+
+## Taxonomy
+- A `Quality Check` is a test that can be performed on SBOM to return a binary result (e.g., A check for specification)
+- A `Quality Check Category` is a logical grouping of Quality Checks (e.g., "NTIA-Minimum-Elements" Checks)
+- A `Quality Check Set` is a collection of Quality Checks (e.g., "Default Check Set", "IoT Quality Set")
+
+## Scoring Methodology
+- Each Quality Check has an equal weight and a score range of 0.0 - 10.0.
```
Comment (riteshnoronha) [inline Features.md:33]: @surendrapathak eventually weights are going to be added which are user customizable. Should we indicate this.. or when its ready

File: Features.md [hunk 2]
```diff
@@ -14,27 +14,308 @@
...
+- A Quality Check applied over a list of items (e.g., licenses) averages its score from the Check applied to each element.
+- Quality Check Set Score is an average of scores over all Quality Checks in that Set.
+
+## Check Set Versioning
+Any Check Set, including the default Check Set, may change over time as new Checks are added, existing ones are removed and meaning of an existing one changes.
+Such a breaking change is marked by incrementing `scoring_engine_version` in the output of `sbomqs`.
+
+Therefore comparing Quality Scores across `scoring_engine_version` is not recommended.
+
+## Default Quality Checks
```
Comment (riteshnoronha) [inline Features.md:43]: Quality Check Sets - Interlynk (Default)

File: Features.md [hunk 3]
```diff
@@ -14,27 +14,308 @@
...
+
+A syntactic error in the SBOM will prevent it from being usable.
+
+***Remediation***
+- Check the SBOM generator tool's known issues and get the most recent version of the tool.
+- Check options/setup of the environment variables required to use the tool.
+- Build SBOM with a different tool.
+
+---
+#### SBOM Specification Syntax
```
Comment (riteshnoronha) [inline Features.md:94]: Duplicate of the above?

File: Features.md [hunk 4]
```diff
@@ -14,27 +14,308 @@
...
+**Remediation**
+
+Identify the component with a missing name and check its product page to get its name.
+- CycloneDX field: [components:name](https://cyclonedx.org/docs/1.4/json/#components_items_name)
+- SPDX field: [PackageName](https://spdx.github.io/spdx-spec/v2.3/package-information/#71-package-name-field)
+
+---
+#### Supplier Name 
+
+This check determines whether each component in the SBOM includes a supplier name.
```
Comment (riteshnoronha) [inline Features.md:123]: We should ack, there is ongoing dicusssions in the open-source ecosystem, about open-source maintainers being classified as suppliers ?

File: Features.md [hunk 5]
```diff
@@ -14,27 +14,308 @@
...
+A declared valid SPDX license is the key to evaluating any compliance risks.
+
+***Remediation steps***
+
+Check and populate the following fields with the relationship of components in the SBOM.
+- CycloneDX field: [component:licenses](https://cyclonedx.org/docs/1.4/json/#components_items_licenses)
+- SPDX fields: [PackageLicenseConcluded](https://spdx.github.io/spdx-spec/v2.3/package-information/#713-concluded-license-field), [LicenseConcluded](https://spdx.github.io/spdx-spec/v2.3/file-information/#85-concluded-license-field)
+
+---
+#### Required Fields
```
Comment (riteshnoronha) [inline Features.md:220]: at this point we only check metadata and components, in future we will expand it to files, etc

**Review by riteshnoronha — CHANGES_REQUESTED** [review]

File: Features.md (hunk 1 above)
Comment (surendrapathak) [reply Features.md:33]: Added the Coming Soon.

File: Features.md (hunk 2 above)
Comment (surendrapathak) [reply Features.md:43]: Fixed

File: Features.md (hunk 3 above)
Comment (surendrapathak) [reply Features.md:94]: It was! 🤦

File: Features.md (hunk 4 above)
Comment (surendrapathak) [reply Features.md:123]: Added

File: Features.md (hunk 5 above)
Comment (surendrapathak) [reply Features.md:220]: I have marked all the Files checks to "Coming Soon"

**Review by riteshnoronha — APPROVED** [review]

**Pull request merged**